ocw_data_parser/course_downloader.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+import json
+import boto3
+
+"""
+This is a class used for downloading source json from S3 based on a list of course id's
+
+An example of the expected format can be found in example_courses.json
+"""
+class OCWDownloader(object):
+    def __init__(self,
+                 courses_json="",
+                 env="QA",
+                 destination_dir="",
+                 s3_bucket_name="",
+                 overwrite=False):
+        self.coursesJson = courses_json
+        self.env = env
+        self.destinationDir = destination_dir
+        self.s3BucketName = s3_bucket_name
+        self.overwrite = overwrite
+
+    def downloadCourses(self):
+        courses = None
+        with open(self.coursesJson) as f:
+            courses = json.load(f)["courses"]
+        if not os.path.exists(self.destinationDir):
+            os.makedirs(self.destinationDir)
+        s3 = boto3.resource("s3")
+        s3Client = boto3.client("s3")
+        ocwContentStorage = s3.Bucket(self.s3BucketName)
+        allObjects = ocwContentStorage.meta.client.list_objects(
+            Bucket=ocwContentStorage.name, Prefix="{}/".format(self.env), Delimiter="/")
+
+        # first tier subfolders under env should be department numbers
+        for departmentPath in allObjects.get("CommonPrefixes"):
```

Comment:
So I think this can be rewritten more cleanly with get_paginator.

You are looping through all the files in the bucket anyway, and just paging through the files sequentially then checking if the file path has the course name you are interested would be easier.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.get_paginator

Reply:
Thanks for this, idk why I didn't come across it when I was looking for a way to loop through all the files in a bucket.